examples/bridge-app/linux/bridged-actions-stub.cpp

```diff
@@ -32,106 +33,231 @@
 {
-    CHIP_ERROR err = aEncoder.EncodeList([&endpoint](const auto & encoder) -> CHIP_ERROR {
-        std::vector<Action *> actionList = GetActionListInfo(endpoint);
+    std::vector<Action *> actionList = GetActionListInfo(mEndpointId);
 
-        for (auto action : actionList)
+    uint16_t visibleCount = 0;
+    for (auto actionPtr : actionList)
+    {
+        if (actionPtr->getIsVisible())
         {
-            if (action->getIsVisible())
+            if (visibleCount == index)
             {
-                Actions::Structs::ActionStruct::Type actionStruct = { action->getActionId(),
-                                                                      CharSpan::fromCharString(action->getName().c_str()),
-                                                                      action->getType(),
-                                                                      action->getEndpointListId(),
-                                                                      action->getSupportedCommands(),
-                                                                      action->getStatus() };
-                ReturnErrorOnFailure(encoder.Encode(actionStruct));
+                // Found the visible action at the requested index
+                action.Set(actionPtr->getActionId(), CharSpan::fromCharString(actionPtr->getName().c_str()), actionPtr->getType(),
+                           actionPtr->getEndpointListId(), actionPtr->getSupportedCommands(), actionPtr->getStatus());
+                return CHIP_NO_ERROR;
             }
+            visibleCount++;
         }
+    }
+
+    return CHIP_ERROR_PROVIDER_LIST_EXHAUSTED;
+}
+
+CHIP_ERROR LinuxActionsDelegateImpl::ReadEndpointListAtIndex(uint16_t index, EndpointListStorage & epList)
+{
+    std::vector<EndpointListInfo> infoList = GetEndpointListInfo(mEndpointId);
+
+    if (index >= infoList.size())
+    {
+        return CHIP_ERROR_PROVIDER_LIST_EXHAUSTED;
+    }
+
+    // Get the endpoint list at the specified index
+    EndpointListInfo info = infoList[index];
+    DataModel::List<const EndpointId> endpointList(info.GetEndpointListData(), info.GetEndpointListSize());
 
-        return CHIP_NO_ERROR;
-    });
-    return err;
+    epList.Set(info.GetEndpointListId(), CharSpan::fromCharString(info.GetName().c_str()), info.GetType(), endpointList);
+
+    return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR ActionsAttrAccess::ReadEndpointListAttribute(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+bool LinuxActionsDelegateImpl::HaveActionWithId(uint16_t actionId, uint16_t & actionIndex)
 {
-    std::vector<EndpointListInfo> infoList = GetEndpointListInfo(endpoint);
+    std::vector<Action *> actionList = GetActionListInfo(mEndpointId);
+
+    for (size_t i = 0; i < actionList.size(); i++)
+    {
+        if (actionList[i]->getActionId() == actionId)
+        {
+            actionIndex = static_cast<uint16_t>(i);
```

Comment:
And does the indexing here need to account for getIsVisible()?

Reply:
Updated the code